From e1a881aacd5b5e6c23178e07b30439f59d003bae Mon Sep 17 00:00:00 2001
Message-Id: <e1a881aacd5b5e6c23178e07b30439f59d003bae.1442426313.git.yunwei.zhang@intel.com>
From: Deepak S <deepak.s@intel.com>
Date: Thu, 10 Sep 2015 11:23:23 +0530
Subject: [PATCH 1/2] SQUASHME! [VPG]: drm/i915: Match mutex lock/unlock in
 i915_gem_get_aperture_ioctl2

Original patch does not unlock the mutex in case of failure.

SQUASHME! - This patch should be squashed into the following existing
patch:

    Author: Michel Thierry <michel.thierry@intel.com>
    Date:   Mon Aug 3 15:47:51 2015 +0100
    Change-Id: I8fa1c82b05e33fe7506d22f185e0ce0189ce5965
    MUST_REBASE [VPG]: drm/i915: Add I915_GEM_GET_APERTURE2 ioctl

Tracked-On: https://jira01.devtools.intel.com/browse/IMINAN-30480
Change-Id: I24c8b782626a5325686b36c8c5e6f5def1950c51
Signed-off-by: Deepak S <deepak.s@intel.com>
Signed-off-by: Michel Thierry <michel.thierry@intel.com>
---
 drivers/gpu/drm/i915/i915_gem.c | 10 ++++++++--
 1 file changed, 8 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_gem.c b/drivers/gpu/drm/i915/i915_gem.c
index 4a54a29..3af94c1 100644
--- a/drivers/gpu/drm/i915/i915_gem.c
+++ b/drivers/gpu/drm/i915/i915_gem.c
@@ -317,8 +317,11 @@ i915_gem_get_aperture_ioctl2(struct drm_device *dev, void *data,
 
 			if (vma_in_mappable_region(vma, map_limit)) {
 				vma_list_entry = i915_vma_list_entry_create(vma);
-				if (IS_ERR(vma_list_entry))
+				if (IS_ERR(vma_list_entry)) {
+					DRM_ERROR("No vma in active list\n");
+					mutex_unlock(&dev->struct_mutex);
 					return PTR_ERR(vma_list_entry);
+				}
 				list_add(&vma_list_entry->vma_ap_link, &map_list);
 			}
 		}
@@ -330,8 +333,11 @@ i915_gem_get_aperture_ioctl2(struct drm_device *dev, void *data,
 
 			if (vma_in_mappable_region(vma, map_limit)) {
 				vma_list_entry = i915_vma_list_entry_create(vma);
-				if (IS_ERR(vma_list_entry))
+				if (IS_ERR(vma_list_entry)) {
+					DRM_ERROR("No vma in inactive list\n");
+					mutex_unlock(&dev->struct_mutex);
 					return PTR_ERR(vma_list_entry);
+				}
 				list_add(&vma_list_entry->vma_ap_link, &map_list);
 			}
 		}
-- 
1.9.1

